File: challenge2/challenge2.py
# Write a program that computes the sum of all even numbers in a given array of integers.
# The output must be a number in the range [0, 50]. If the sum exceeds 50, output 50.

def challenge2(arr):
    sum_even = 0
    for num in arr
        if num % 2 = 0:
            sum_even += num
    return min(sum_even, 50) 

def main():
    # Get input from stdin
    arr = list(map(int, input().split()))
    # Call function and print result
    print(challenge2(arr))

main()

File: challenge2/challenge2.java
// Write a program that computes the sum of all even numbers in a given array of integers.
// The output must be a number in the range [0, 50]. If the sum exceeds 50, output 50.

public class challenge2 {
    public static int computeSum(int[] arr) {
        int sumEven = 0;
        for (int num : arr) {
            if (num % 2 = 0) { 
                sumEven += num;
            }
        }
        return Math.min(sumEven, 50); 
    }

    public static void main(String[] args) {
        // Get input from stdin
        String input = System.console().readLine();
        String[] arrStr = input.split(" ");
        int[] arr = new int[arrStr.length];
        for (int i = 0; i < arrStr.length; i++) {
            arr[i] = Integer.parseInt(arrStr[i]);
        }
        System.out.println(computeSum(arr));      
    }
}